core/shared/src/main/scala/zio/prelude/AssociativeEither.scala

```diff
@@ -222,6 +240,15 @@ object AssociativeEither extends LawfulF.Invariant[AssociativeEitherDeriveEqualI
         fa.map(Left(_)) orElse fb.map(Right(_))
     }
 
+  /**
+   * The `CommutativeEither` instance for `ZStream`.
+   */
+  implicit def ZStreamCommutativeEither[R, E]: CommutativeEither[({ type lambda[+a] = ZStream[R, E, a] })#lambda] =
+    new CommutativeEither[({ type lambda[+a] = ZStream[R, E, a] })#lambda] {
+      def either[A, B](fa: => ZStream[R, E, A], fb: => ZStream[R, E, B]): ZStream[R, E, Either[A, B]] =
+        fa mergeEither fb
+    }
+
   /**
    * The `AssociativeEither` instance for failed `ZStream`.
```

Comment:
`ZStreamFailureAssociativeEither` should be removed

Reply:
@sideeffffect what about with `ZIOFailureAssociativeEither` also should be removed?

Reply:
`ZIO[R, E, +*]` and `Failure[ZIO[R, +*, A]]` are different types, so both can stay